disperser/apiserver/server.go

```diff
@@ -88,468 +78,6 @@ func NewDispersalServer(
-	in, err := stream.Recv()
-	if err != nil {
-		return fmt.Errorf("error receiving next message: %v", err)
-	}
-
-	request, ok := in.Payload.(*pb.AuthenticatedRequest_DisperseRequest)
-	if !ok {
-		return errors.New("expected DisperseBlobRequest")
-	}
-
-	blob := getBlobFromRequest(request.DisperseRequest)
-
-	// Get the ethereum address associated with the public key. This is just for convenience so we can put addresses instead of public keys in the allowlist.
-	// Decode public key
-	publicKeyBytes, err := hexutil.Decode(blob.RequestHeader.AccountID)
-	if err != nil {
-		return fmt.Errorf("failed to decode public key (%v): %v", blob.RequestHeader.AccountID, err)
-	}
-
-	pubKey, err := crypto.UnmarshalPubkey(publicKeyBytes)
-	if err != nil {
-		return fmt.Errorf("failed to decode public key (%v): %v", blob.RequestHeader.AccountID, err)
-	}
-
-	authenticatedAddress := crypto.PubkeyToAddress(*pubKey).String()
-
-	// Send back challenge to client
-	challenge := rand.Uint32()
-	err = stream.Send(&pb.AuthenticatedReply{Payload: &pb.AuthenticatedReply_BlobAuthHeader{
-		BlobAuthHeader: &pb.BlobAuthHeader{
-			ChallengeParameter: challenge,
-		},
-	}})
-	if err != nil {
-		return err
-	}
-
-	// Recieve challenge_reply
-	in, err = stream.Recv()
-	if err != nil {
-		return fmt.Errorf("error receiving next message: %v", err)
-	}
-
-	challengeReply, ok := in.Payload.(*pb.AuthenticatedRequest_AuthenticationData)
-	if !ok {
-		return errors.New("expected AuthenticationData")
-	}
-
-	blob.RequestHeader.Nonce = challenge
-	blob.RequestHeader.AuthenticationData = challengeReply.AuthenticationData.AuthenticationData
-
-	err = s.authenticator.AuthenticateBlobRequest(blob.RequestHeader.BlobAuthHeader)
-	if err != nil {
-		return fmt.Errorf("failed to authenticate blob request: %v", err)
-	}
-
-	// Disperse the blob
-	reply, err := s.disperseBlob(stream.Context(), blob, authenticatedAddress)
-	if err != nil {
-		s.logger.Info("failed to disperse blob", "err", err)
-		return err
-	}
-
-	// Send back disperse_reply
-	err = stream.Send(&pb.AuthenticatedReply{Payload: &pb.AuthenticatedReply_DisperseReply{
-		DisperseReply: reply,
-	}})
-	if err != nil {
-		s.logger.Error("failed to stream back DisperseReply", "err", err)
-		return err
-	}
-
-	return nil
-
-}
-
-func (s *DispersalServer) DisperseBlob(ctx context.Context, req *pb.DisperseBlobRequest) (*pb.DisperseBlobReply, error) {
-
-	blob := getBlobFromRequest(req)
-
-	reply, err := s.disperseBlob(ctx, blob, "")
-	if err != nil {
-		s.logger.Info("failed to disperse blob", "err", err)
-	}
-	return reply, err
-}
-
-func (s *DispersalServer) disperseBlob(ctx context.Context, blob *core.Blob, authenticatedAddress string) (*pb.DisperseBlobReply, error) {
-	timer := prometheus.NewTimer(prometheus.ObserverFunc(func(f float64) {
-		s.metrics.ObserveLatency("DisperseBlob", f*1000) // make milliseconds
-	}))
-	defer timer.ObserveDuration()
-
-	securityParams := blob.RequestHeader.SecurityParams
-	if len(securityParams) == 0 {
```

Comment:
These validation errors aren't currently resulting in a call to `s.metrics.HandleFailedRequest(quorumId, blobSize, "DisperseBlob")`